hudi-spark/src/main/java/org/apache/hudi/DataSourceUtils.java

```diff
@@ -218,6 +218,13 @@ public static HoodieRecord createHoodieRecord(GenericRecord gr, Comparable order
     return new HoodieRecord<>(hKey, payload);
   }
 
+  /**
+   * Drop duplicate records, whose location (file group/file id) mapped by index exists.
```

Comment:
we could make this more descriptive. "Drop duplicate records from incoming records. Drop those records that are already present in the dataset"

Reply:
Done, PTAL

Reply:
Hi  @nsivabalan. I have addressed all of your comments, please take another look